esphome/components/adc/adc_sensor.h

```diff
@@ -34,7 +34,7 @@ class ADCSensor : public sensor::Sensor, public PollingComponent, public voltage
   uint8_t pin_;
 
 #ifdef ARDUINO_ARCH_ESP32
-  adc_attenuation_t attenuation_{ADC_0db};
+  adc_atten_t attenuation_{};
```

Comment:
Nit: `ADC_ATTEN_DB_0` is 0 in fact, but especially since the attenuation setter might not get called it feels cleaner to explicitly initialize.

```suggestion
  adc_atten_t attenuation_{ADC_ATTEN_DB_0};
```